lib/SILOptimizer/Transforms/AccessEnforcementOpts.cpp

```diff
@@ -401,146 +575,231 @@ void AccessConflictAnalysis::identifyBeginAccesses() {
   }
 }
 
-/// When a conflict is detected, flag the access so it can't be folded, and
-/// remove its index from the current access set so we stop checking for
-/// conflicts. Erasing from SparseAccessSet does not invalidate any iterators.
-static void recordConflict(AccessInfo &info, SparseAccessSet &accessSet) {
-  info.setSeenNestedConflict();
-  accessSet.setConflict(info.getAccessIndex());
+void RegionStorageAnalysis::identifyStorageLocations() {
+  for (auto it : result.accessMap) {
+    auto storage = it.second;
+    storageAccesses.insert(storage);
+  }
+}
+
+void RegionStorageAnalysis::calcBottomUpOrder(
```

Comment:
A comment would be helpful: "Returns a worklist of loop IDs is bottom-up order."